crates/types/src/identifiers.rs

```diff
@@ -855,144 +793,165 @@ impl FromStr for LambdaARN {
     }
 }
 
-#[derive(
-    PartialOrd,
-    PartialEq,
-    Eq,
-    Hash,
-    Clone,
-    Copy,
-    serde_with::SerializeDisplay,
-    serde_with::DeserializeFromStr,
-)]
-pub struct PartitionProcessorRpcRequestId(Ulid);
-
-impl PartitionProcessorRpcRequestId {
-    pub fn new() -> Self {
-        Self(Ulid::new())
-    }
+/// Generate an identifier backed by ULID.
+///
+/// This generates the struct macro plus bunch of common methods. To use:
```

Comment:
Maybe remove this sentence. W/o specifying what methods are added, I don't think that this adds value.